lib/dataset/test/bin_test.cpp

```diff
@@ -198,6 +198,21 @@ TEST_P(BinTest, rebin_using_attr) {
   EXPECT_EQ(expected, result);
 }
 
+TEST_P(BinTest, rebin_using_attr_in_new_dimension) {
+  auto table = GetParam();
+  auto expected_temp = bin(table, {edges_x});
+  const auto z_coord = makeVariable<double>(Dims{Dim::X}, Shape{4},
+                                            Values{-10., -5.0, 0.5, 7.5});
+  expected_temp.coords().set(Dim::Z, z_coord);
+  const auto edges_z_coarse =
+      makeVariable<double>(Dims{Dim::Z}, Shape{3}, Values{-11., 0.0, 8.0});
+  const auto expected = bin(expected_temp, {edges_z_coarse});
+  auto temp = bin(table, {edges_x});
+  temp.attrs().set(Dim::Z, z_coord);
```

Comment:
So this is the same as `expected_temp` but with `Dim::Z` as an attr rather than coord? Can we use some clearer variable names?

It may also help to interleave the operations on the two, to highlight the parallel/identical actions, something like:

```cpp
auto da_coord = bin(table, {edges_x});
auto da_attr = bin(table, {edges_x});
da_coord.coords().set(Dim::Z, z_coord);
da_attr.attrs().set(Dim::Z, z_coord);
const auto out_coord = bin(da_coord, {edges_z_coarse});
const auto out_attr = bin(da_attr, {edges_z_coarse});
```
?